tests/test_metadata_integration.py

```diff
@@ -0,0 +1,269 @@
+            run_dirs = [
+                d
+                for d in metadata_dir.iterdir()
+                if d.is_dir() and d.name.startswith("run_")
+            ]
+            assert (
+                len(run_dirs) == 1
+            ), f"Should have exactly one run directory, found: {run_dirs}"
+
+            run_dir = run_dirs[0]
+
+            # Check for entity-specific subdirectories with entity names
+            model_dir = run_dir / "model"
+            ensemble_dir = run_dir / "ensemble" / "test_ensemble"
+            database_dir = run_dir / "database"
+
+            assert (
+                not model_dir.exists()
+            ), f"Model metadata directory should not exist: {model_dir}"
+            assert (
+                ensemble_dir.exists()
+            ), f"Ensemble metadata directory should exist: {ensemble_dir}"
+            assert (
+                not database_dir.exists()
+            ), f"Database metadata directory should not exist: {database_dir}"
+
+            # Clean up
+            exp.stop(ensemble)
+
+    def test_experiment_creates_correct_metadata_directory_structure_all_types(self):
+        """Test that launching models, ensembles, and orchestrator creates all directories"""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            exp = Experiment("test_metadata_all", exp_path=temp_dir, launcher="local")
+
+            # Create model
+            model = exp.create_model(
+                "test_model", run_settings=exp.create_run_settings("echo", ["hello"])
+            )
+
+            # Create ensemble
+            ensemble = exp.create_ensemble(
+                "test_ensemble",
+                run_settings=exp.create_run_settings("echo", ["world"]),
+                replicas=2,
+            )
+
+            exp.start(model, ensemble, block=True)
+
+            # Verify directory structure
+            smartsim_dir = pathlib.Path(temp_dir) / CONFIG.smartsim_base_dir
+            metadata_dir = smartsim_dir / "metadata"
+
+            assert metadata_dir.exists(), "Metadata directory should exist"
+
+            # Check for run-specific subdirectories (single launch, so single run dir)
+            run_dirs = [
+                d
+                for d in metadata_dir.iterdir()
+                if d.is_dir() and d.name.startswith("run_")
+            ]
+            assert (
+                len(run_dirs) == 1
+            ), f"Should have exactly one run directory, found: {run_dirs}"
+
+            run_dir = run_dirs[0]
+
+            # Check for entity-specific subdirectories with entity names
+            model_dir = run_dir / "model" / "test_model"
+            ensemble_dir = run_dir / "ensemble" / "test_ensemble"
+
+            assert (
+                model_dir.exists()
+            ), f"Model metadata directory should exist: {model_dir}"
+            assert (
+                ensemble_dir.exists()
+            ), f"Ensemble metadata directory should exist: {ensemble_dir}"
```

Comment:
Just checking because it was added in other tests: do we want this check here also?
```suggestion
            assert (
                model_dir.exists()
            ), f"Model metadata directory should exist: {model_dir}"
            assert (
                ensemble_dir.exists()
            ), f"Ensemble metadata directory should exist: {ensemble_dir}"
            assert (
                not database_dir.exists()
            ), f"Database metadata directory should not exist: {database_dir}"
```